### PR TITLE
[#9593] Add SauceLabs integration

### LINKED ISSUE #9593: Add SauceLabs integration
Now that we have had some minimal E2E tests set up, it is about time to set up the integration for SauceLabs for cross-browser testing.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #9593 

- Support `saucelabs` as one of the acceptable browsers in our E2E tests
- Support running the E2E tests remotely on SauceLabs servers for each `master` branch push (not PR) if there are `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` env variables set.

~Caveat:~

~This will break external forks' `master` branch build in Travis (if any are set up) because they do not have the valid username/access key pair to run E2E tests on SauceLabs servers (our Open Sauce account obviously cannot be shared with any contributor).~

### CODE REVIEW

**Comment by wkurniawan07** [comment]
While @xpdavid will be the reviewer, I'll tag @jacoblipech and @AyushChatto along to pass down some knowledge.

File: src/e2e/java/teammates/e2e/pageobjects/BrowserPool.java [hunk 1]
```diff
@@ -39,19 +43,27 @@ private static BrowserPool getInstance() {
...
      */
-    public static void release(Browser browser) {
-        BrowserPool pool = getInstance();
-        // Synchronized to ensure thread-safety
-        synchronized (pool) {
-            browser.isInUse = false;
-            pool.notifyAll();
+    public static void release(Browser browser, boolean isSuccess) {
+        if (TestProperties.BROWSER_SAUCELABS.equals(TestProperties.BROWSER)) {
+            ((JavascriptExecutor) browser.driver).executeScript("sauce:job-result=" + (isSuccess ? "passed" : "failed"));
```
Comment (xpdavid) [inline src/e2e/java/teammates/e2e/pageobjects/BrowserPool.java]: I guess here we want to notify sauce lab about the result of the execution of the test?

Maybe we want to put this in https://github.com/TEAMMATES/teammates/pull/9597/files#diff-5ac8e80833810cd52d80b7f75531503dR62 instead. After all, browser should have no idea about whether a test is success or not (so it is a bit confusing to have a `isSuccess` parameter in the method signature)?

File: src/e2e/resources/test.saucelabs.properties [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#-----------------------------------------------------------------------------
+# This file contains specific configuration values for testing on Travis.
```
Comment (xpdavid) [inline src/e2e/resources/test.saucelabs.properties]:
```suggestion
# This file contains specific configuration values for testing on Travis against Sauce Lab.
```

File: e2e.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
```
Comment (xpdavid) [inline e2e.sh]: In addition, I think it is better to name the file to something like `runE2EAgainstSauceLab.sh` cause it is specific to source lab?

**Comment by wkurniawan07** [comment]
While this is ready for merging, putting this on hold until we have more E2E tests.

**Comment by wkurniawan07** [comment]
Will be closing this for now. We're more better positioned to actually make use of Saucelabs now, but the integration with GitHub Actions needs to studied first.

**Pull request closed without merging**